src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,146 @@
+import { Component, Prop, h, Event, EventEmitter, State } from '@stencil/core';
+import { EditorTextLink } from '../prosemirror-adapter/menu/types';
+import { Languages } from 'src/components/date-picker/date.types';
+import translate from '../../../global/translations';
+
+/**
+ * This component is a menu for editing a link in the text editor.
+ * It allows the user to input a title of a link and a link itself.
+ */
+
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
```

Comment:
State-props are more internal than events, so we place state props after the event emitter properties. (I'll fix this, just wanted to comment to spread the knowledge 😊)